### PR TITLE
feat: Remove account lockout

### LINKED ISSUE #146: Remove the account lockout feature
With signin confirmation and email caphta we have more targeted
meant to keep bad people at bay while minimizing friction for valid
users, account lockout no longer seems particularly elegant.

Ref https://github.com/mozilla/fxa-content-server/issues/3949
Ref https://github.com/mozilla/fxa-auth-server/issues/1359

### PR DESCRIPTION
**Pull request by shane-tomlinson**
With signin confirmation and email caphta we have more targeted
meant to keep bad people at bay while minimizing friction for valid
users, account lockout no longer seems particularly elegant.

fixes #146

Should be merged after https://github.com/mozilla/fxa-auth-server/pull/1360

### CODE REVIEW

**Comment by shane-tomlinson** [comment]
This PR removes the methods and tests, but does not make any changes to the actual DB or drop any stored procedures.

File: lib/db/mem.js [hunk 1]
```diff
@@ -673,7 +670,6 @@ module.exports = function (log, error) {
       this.deletePasswordForgotToken(tokenId),
       this.createAccountResetToken(accountResetToken.tokenId, accountResetToken),
       this.verifyEmail(accountResetToken.uid),
```
Comment (seanmonstar) [inline lib/db/mem.js]: Dangling comma is still a syntax error in JS, right?

**Comment by seanmonstar** [comment]
Deletions woooo!

Why the decision to not remove the stored procedures?

**Comment by rfk** [comment]
> Why the decision to not remove the stored procedures?

In general we don't, because we might need them if rolling back to an earlier version of the server.  Probably not an issue in practice with this feature since it was never enabled in prod.  We should consider a follow-up to actually drop the things we don't use in the db.

**Comment by shane-tomlinson** [comment]
> We should consider a follow-up to actually drop the things we don't use in the db.

I'm happy to do either one.

**Comment by vladikoff** [comment]
We also have https://github.com/mozilla/fxa-auth-db-mysql/issues/142 that requires some DB changes I think. I wonder if it makes sense to do both in the same migration or not.

**Comment by rfk** [comment]
> I wonder if it makes sense to do both in the same migration or not.

I like the idea of a single "remove stuff we're not using any more" migration to clobber both of these things

**Comment by shane-tomlinson** [comment]
So what's the resolution on this? Keep the stored procedures and drop in one batch?

**Comment by rfk** [comment]
Yes, leave the DB alone and note it in a follow-up bug

**Comment by shane-tomlinson** [comment]
> Yes, leave the DB alone and note it in a follow-up bug

If that's the case, @seanmonstar - mind an r?

**Pull request merged**